skyportal/models.py

```diff
@@ -300,9 +316,11 @@ def altitude(self, telescope, time):
 
 
 class Filter(Base):
-    query_string = sa.Column(sa.String, nullable=False, unique=False)
+    name = sa.Column(sa.String, nullable=False, unique=False)
+    stream_id = sa.Column(sa.ForeignKey("streams.id"), nullable=False)
+    stream = relationship("Stream", foreign_keys=[stream_id], back_populates="filters")
     group_id = sa.Column(sa.ForeignKey("groups.id"), nullable=False)
```

Comment:
```suggestion
    group_id = sa.Column(sa.ForeignKey("groups.id"), nullable=False, ondelete='CASCADE')
```

Reply:
Nice catch, fixed

Reply:
Also make sure to index all foreign keys with `index=True`, to ensure fast joins.

Reply:
Done.